src/modules/app.ts

```diff
@@ -339,18 +339,21 @@ export class AppModule
    * @returns - The sanitized metadata with an uploaded image ipfs hash
    */
   private async _prepareMetadata(metadata: CommonModuleMetadata): Promise<any> {
-    if (typeof metadata.image === "string") {
-      return Promise.resolve(metadata);
-    }
-    if (metadata.image === undefined) {
-      return Promise.resolve(metadata);
+    for(const key in metadata) {
+      if (typeof metadata[key] === "string") { 
+        return Promise.resolve(metadata);
+      }
+      if (metadata[key] === undefined) { 
+        return Promise.resolve(metadata);
+      }
+  
+      metadata[key] = await uploadToIPFS(
```

Comment:
This is not explicit enough. You need to make sure its a file before uploading it.